### ISSUE TITLE
Allocation error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sjaaky74/netcdf4-python#53]
```
What is the expected output? What do you see instead?

I'm copying data from a pytables array & table into a netcdf4-python dataset 
for comparison 
testing. The pytables database contains about 1.3 million 461x3 numpy arrays 
(single float). 
When copying the data into the netcdf dataset, after about 500,000 elements I 
get the following 
error:

Python(574) malloc: *** mmap(size=16777216) failed (error code=12)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
Python(574) malloc: *** mmap(size=16777216) failed (error code=12)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
Python(574) malloc: *** mmap(size=16777216) failed (error code=12)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
Python(574) malloc: *** mmap(size=16777216) failed (error code=12)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
Traceback (most recent call last):
  File "convPyTable2netcdf4.py", line 41, in <module>
    ncfile.variables['coords'][segid,:,:] = hf5.root.x[k]
  File "netCDF4.pyx", line 1862, in netCDF4.Variable.__setitem__ (netCDF4.c:10782)
  File "netCDF4.pyx", line 1984, in netCDF4.Variable._put (netCDF4.c:11560)
RuntimeError: HDF error

See attached code for implementation. The pytables database is about 6.8 GB and 
the netcdf4 file 
is 2.8 GB when the error occurs. 

What version of the product are you using? On what operating system?
I'm using EPD 6.1 (netcdf4-python v0.7.6) on a Macbook Pro running OS 10.6.3 
with 4 GB RAM.

Please provide any additional information below.


```

Original issue reported on code.google.com by `joshua.a...@gmail.com` on 5 Jun 2010 at 12:47

Attachments:
- [convPyTable2netcdf4.py](https://storage.googleapis.com/google-code-attachments/netcdf4-python/issue-53/comment-0/convPyTable2netcdf4.py)

**Comment by GoogleCodeExporter** [comment]
```
Just to add some additional information to my original issue, since I realized 
it would be hard to verify since I 
don't supply the pytables-generated file, I can reproduce the error by simply 
trying to fill a netcdf4 file directly 
using a numpy array (see attached file).


```

Original comment by `joshua.a...@gmail.com` on 5 Jun 2010 at 1:43

Attachments:
- [convPyTable2netcdf4_test.py](https://storage.googleapis.com/google-code-attachments/netcdf4-python/issue-53/comment-1/convPyTable2netcdf4_test.py)

**Comment by GoogleCodeExporter** [comment]
```
Building netcdf4-python v0.9.1 and v0.7.6 from scratch with a fresh Python 
2.6.5 build, I cannot reproduce the 
error that I see using the Enthough Python Distribution which uses v0.7.6, so 
it looks like an issue with EPD and 
not with netcdf4-python. I'll contact the EPD developers and let them know. If 
anyone has additional insight 
about what might be going on though, please let me know.
```

Original comment by `joshua.a...@gmail.com` on 6 Jun 2010 at 8:21

**Comment by GoogleCodeExporter** [comment]
```
Must have been a bug I fixed between 0.7.6 and 0.9.1. 

-Jeff
```

Original comment by `whitaker.jeffrey@gmail.com` on 9 Jun 2010 at 3:47

**Comment by GoogleCodeExporter** [comment]
Original comment by `whitaker.jeffrey@gmail.com` on 26 Feb 2014 at 2:04
- Changed state: **Fixed**
